### PR TITLE
3170 informed consent

### PR DESCRIPTION
**Pull request by pfriedman**
This fairly large change addresses issues with Informed Consent. Several of these changes are already being tested in various environments as at times the head of this branch has been deployed. Things being tested are Reconsent, Withdrawal, and Child Consent.

There are several conflicts between this branch and the current master and a rebase or patch will need to happen before this PR can even be considered to be merged.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -50,6 +50,10 @@ def update
         mark_activity_occurred
         @event.update_associated_informed_consent_event
 
+        if @event.screener_event?
+          EligibilityAdjudicator.adjudicate_eligibility(@event.participant.person)
+        end
+
```
Comment (jdzak) [inline app/controllers/events_controller.rb]: Just a reminder to move this to a place where Field can also benefit from this.

Comment (hannahwhy) [inline app/controllers/events_controller.rb]: Agreed.

On a related note, I'd like to know why useful business logic keeps getting stuck in controllers.  (A lot of this `update` action seems like it really needs to be pushed to the model layer.)  I thought it was clear that such code can no longer be confined to a single request.

**Comment by pfriedman** [comment]
As much of this PR is already in testing on the QA servers and as it is needed for the 1.5.0 release candidate I am merging this PR

File: vendor/ncs_navigator_schema [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit a2bcbe67bcb06fc32e93165e7cd45b95beede417
+Subproject commit c9b696297c7d857a492be9c04b55afeb65e0d7ce
```
Comment (jdzak) [inline vendor/ncs_navigator_schema:1]: Watch out here, this should be c9b696297c7d857a492be9c04b55afeb65e0d7ce.

Comment (jdzak) [inline vendor/ncs_navigator_schema:1]: oops, other way around.. a2bcbe67bcb06fc32e93165e7cd45b95beede417

**Pull request merged**